libgnucash/engine/Account.cpp

```diff
@@ -3403,14 +3403,16 @@ gnc_numeric
 xaccAccountGetReconciledBalanceAsOfDate (Account *acc, time64 date)
 {
     gnc_numeric balance = gnc_numeric_zero();
+    time64 today = gnc_time64_get_today_end ();
 
     g_return_val_if_fail(GNC_IS_ACCOUNT(acc), gnc_numeric_zero());
 
     for (GList *node = GET_PRIVATE(acc)->splits; node; node = node->next)
     {
         Split *split = (Split*) node->data;
+        time64 reconciled_date = xaccSplitGetDateReconciled (split);
         if ((xaccSplitGetReconcile (split) == YREC) &&
-            (xaccSplitGetDateReconciled (split) <= date))
+            ((reconciled_date <= date) || (reconciled_date > today)))
```

Comment:
I don't like this one at all: It means that every past-reconcile will have the future reconcile errors attached to its balance. I think that will make it even harder to find and fix the bad reconcile dates.